unittests/FEXLinuxTests/CMakeLists.txt

```diff
@@ -1,31 +1,33 @@
 include(ExternalProject)
-ExternalProject_Add(FEXLinuxTests
-  PREFIX FEXLinuxTests
-  SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/tests"
-  BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/FEXLinuxTests"
-  CMAKE_ARGS
-  "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
-  "-DCMAKE_TOOLCHAIN_FILE:FILEPATH=${X86_TOOLCHAIN_FILE}"
-  INSTALL_COMMAND ""
-  BUILD_ALWAYS ON
-  )
 
-ExternalProject_Add(FEXLinuxTests_32
-  PREFIX FEXLinuxTests_32
-  SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/tests-32"
-  BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/FEXLinuxTests_32"
-  CMAKE_ARGS
-  "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
-  "-DCMAKE_TOOLCHAIN_FILE:FILEPATH=${X86_TOOLCHAIN_FILE}"
-  INSTALL_COMMAND ""
-  BUILD_ALWAYS ON
-  )
+set(GuestArchList "x86_32;x86_64")
+
+foreach(GUEST_ARCH ${GuestArchList})
+  set(TOOLCHAIN_FILE ${TOOLCHAIN_FILE_${GUEST_ARCH}})
+
+  # make sure configure is run again if the toolchain file is modified
+  set_directory_properties(PROPERTIES CMAKE_CONFIGURE_DEPENDS "${TOOLCHAIN_FILE}")
```

Comment:
AFAICT this won't catch modifications to the `-generic` toolchain file.

Reply:
seriously? doh cmake.

Reply:
Fixed